server/routes/archiveGoal/archiveGoalFormValidator.test.ts

```diff
@@ -0,0 +1,84 @@
+import type { ArchiveGoalForm } from 'forms'
+import aValidArchiveGoalForm from '../../testsupport/archiveGoalFormTestDataBuilder'
+import validateArchiveGoalForm from './archiveGoalFormValidator'
+import ReasonToArchiveGoalValue from '../../enums/ReasonToArchiveGoalValue'
+
+describe('archiveGoalFormValidator', () => {
+  it('should validate given no errors', () => {
+    // Given
+    const form: ArchiveGoalForm = aValidArchiveGoalForm()
+
+    // When
+    const errors = validateArchiveGoalForm(form)
+
+    // Then
+    expect(errors).toEqual([])
+  })
+
+  it('should require a reason to be selected', () => {
+    // Given
+    const form: ArchiveGoalForm = {
+      reference: '95b18362-fe56-4234-9ad2-11ef98b974a3',
+      title: 'Learn French',
+    }
+
+    // When
+    const errors = validateArchiveGoalForm(form)
+
+    // Then
+    expect(errors).toEqual([{ href: '#reason', text: 'You must select a reason to archive the goal' }])
+  })
+
+  it.each([undefined, ''])('should require details when selecting OTHER', anInvalidReasonOther => {
+    // Given
+    const form: ArchiveGoalForm = {
+      reference: '95b18362-fe56-4234-9ad2-11ef98b974a3',
+      title: 'Learn French',
+      reason: ReasonToArchiveGoalValue.OTHER,
+      reasonOther: anInvalidReasonOther,
+    }
+
+    // When
+    const errors = validateArchiveGoalForm(form)
+
+    // Then
+    expect(errors).toEqual([{ href: '#reasonOther', text: 'Enter the reason you are archiving the goal' }])
+  })
+
+  it('other reason should be 200 chars or less', () => {
+    // Given
+    const anInvalidReasonOther = new Array(202).join('x')
```

Comment:
I'm confused by this test.
It looks like we setup the data with the answer for the OTHER field as 202 characters in length:
```
const anInvalidReasonOther = new Array(202).join('x')
```
But then the error message suggests we entered 201 characters?
```
    expect(errors).toEqual([
      { href: '#reasonOther', text: 'The reason is limited to 200 characters but you entered 201' },
    ])
```

Reply:
Would `const anInvalidReasonOther = 'x'.repeat(201)` be clearer ? 🤔

Reply:
It's a weird nuance of join. We are joining 202 empty elements with x so you don't get one for the first/last element, e.g., <empty>x<empty>x<empty> is a list of 3 that produce 2 x's. I've realised I can replace it with padString which is a bit simpler to read.